doc/src/devdocs/ast.md

```diff
@@ -106,7 +106,7 @@ Doc string syntax:
 f(x) = x
 ```
 
-parses as `(macrocall (|.| Core '@doc) (line) "some docs" (= (call f x) (block x)))`.
+parses as `(macrocall GlobalRef(Core, Symbol("@doc")) (line) "some docs" (= (call f x) (block x)))`.
```

Comment:
Since we use s-expressions here, I believe this would be the correct syntax:
```suggestion
parses as `(macrocall (core @doc) (line) "some docs" (= (call f x) (block x)))`.
```

Reply:
The S-Expr were based on each expression being an Expr, and I'm not sure how to fit this into that syntax. The way you've written it I'd interpret as `Expr(:core, :(@doc))`, which isn't right.

Reply:
That's how it's represented in the Flisp parser and in lowering. `core`, `top` and `globalref` expressions are then converted to `GlobalRef` only once it's translated back to Julia, which is implemented in `ast.c`. Flisp also doesn't special-case `@` in front of identifiers, so `@doc` is just a regular identifier, not a macrocall.

Reply:
Okay, but that info isn't in the docs about the Julia AST and `core`, `top` and `globalref` are not mentioned as expression types. I think that ideally I should be able to read the Metaprogramming chapter and this section about the Julia AST and learn everything I need for handling the output of `Meta.parse` or for writing macros.

If we do document the `core`, `top` and `globalref` expressions, that would still not be enough, imo, because from the docs an implementor would still be expecting to receive `Expr`, not the lowered `GlobalRef` types.

Reply:
And thanks for explaining!

Reply:
I think you are right, https://docs.julialang.org/en/v1/devdocs/ast/#Lowered-form should probably mention how these Julia types are represented in the Lisp parser. My impression was that these devdocs are meant mainly for people working on Julia's compiler frontend, which is the reason all the examples write the AST as it is represented in Lisp, so I think we should be consistent here. Maybe we should add some more on ASTs to the manual's main metaprogramming section though and gear that more towards metaprogramming in Julia than working on internals.